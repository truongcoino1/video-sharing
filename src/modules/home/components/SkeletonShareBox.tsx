import React, { memo } from "react";
import Skeleton from "react-loading-skeleton";

const SkeletonShareBox = (): JSX.Element => {
  return (
    <div className="sharebox-container mt-[100px] flex items-center justify-center">
      <Skeleton className="w-[600px] min-w-[600px] min-h-[178px]" containerClassName="w-[600px] min-w-[600px] min-h-[178px] rounded-large" />
    </div>
  );
};

export default memo(SkeletonShareBox);
