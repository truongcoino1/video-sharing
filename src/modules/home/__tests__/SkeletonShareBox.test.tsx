import { render, cleanup } from "@testing-library/react";
import SkeletonShareBox from "../components/SkeletonShareBox";

const push = jest.fn();
jest.mock("next/navigation", () => ({
  useRouter: () => ({
    push: push,
  }),
  usePathname: () => "/",
}));

jest.mock("../hooks/useMovies", () => {
  return {
    useMovies: jest.fn().mockReturnValue({
      movies: [],
      getMovies: jest.fn(),
      period: {
        page: 0,
        limited: false,
        pageSize: 20,
        refresh: false,
        loadMore: false,
      },
      setMovies: jest.fn(),
    }),
  };
});


describe("SkeletonShareBox", () => {
  afterEach(cleanup);

  it("share should render without crash", () => {
    const result = render(<SkeletonShareBox />);
    const elm = result.container.querySelector(".sharebox-container-skeleton");
    expect(elm).toBeInTheDocument();
  });

  it("should render correct SkeletonMovie", () => {
    const element = render(<SkeletonShareBox />);
    expect(element).toMatchSnapshot();
  });
});
