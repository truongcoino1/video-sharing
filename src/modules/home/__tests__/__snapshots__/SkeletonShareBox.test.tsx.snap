// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonShareBox should render correct SkeletonMovie 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sharebox-container-skeleton mt-[100px] flex items-center justify-center"
      >
        <span
          aria-busy="true"
          aria-live="polite"
          class="w-[600px] min-w-[600px] min-h-[178px] rounded-large"
        >
          <span
            class="react-loading-skeleton w-[600px] min-w-[600px] min-h-[178px]"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sharebox-container-skeleton mt-[100px] flex items-center justify-center"
    >
      <span
        aria-busy="true"
        aria-live="polite"
        class="w-[600px] min-w-[600px] min-h-[178px] rounded-large"
      >
        <span
          class="react-loading-skeleton w-[600px] min-w-[600px] min-h-[178px]"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
